port/osx/omrosdump.c

```diff
@@ -0,0 +1,388 @@
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup Port
+ * @brief User space system dump for OSX
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <mach/mach.h>
+#include <mach/mach_vm.h>
+#include <mach/machine.h>
+#include <mach/thread_status.h>
+#include <mach/vm_region.h>
+#include <mach-o/loader.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/proc.h>
+#include <sys/sysctl.h>
+#include <sys/syslimits.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "omrport.h"
+#include "omrportpriv.h"
+#include "omrosdump_helpers.h"
+
+/* current thread flavors needed for OSX debugging */
+struct thread_command_full_64 {
+	uint32_t cmd;
+	uint32_t cmdsize;
+	x86_thread_state_t thread_state;
+	x86_float_state_t float_state;
+	x86_exception_state_t exceptions;
+};
+
+static char corefile_name[PATH_MAX];
+static int corefile_fd = -1;
+
+static int coredump_to_file(mach_port_t, pid_t);
+static int list_thread_commands(mach_port_t, struct thread_command_full_64 **, natural_t *);
+static int list_segment_commands(mach_port_t, struct segment_command_64 **, natural_t *);
+
+static int coredump_to_file(mach_port_t task_port, pid_t pid) {
+	kern_return_t kr = KERN_SUCCESS;
+	struct mach_header_64 *mh64;
+	struct segment_command_64 *segments = NULL;
+	struct thread_command_full_64 *threads = NULL;
+	natural_t segment_count;
+	natural_t thread_count;
+	natural_t cpu_count;
+	processor_basic_info_t proc_info_array;
+	mach_msg_type_number_t info_count;
+	off_t file_off = 0;
+	uint64_t seg_file_off = 0;
+
+	corefile_fd = open(corefile_name, O_RDWR | O_CREAT | O_EXCL, 0600);
+
+	if (-1 == corefile_fd) {
+		perror("open()");
+		goto done;
+	}
+
+	kr = host_processor_info(mach_host_self(), PROCESSOR_BASIC_INFO, &cpu_count, (processor_info_array_t *)&proc_info_array, &info_count);
```

Comment:
This line (and some others) is too long, especially if it can't be completely seen in the web UI.